### PR TITLE
Backport toEnd optimisations to 2.23

### PR DESCRIPTION
**Pull request by tgd**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java [hunk 1]
```diff
@@ -1017,6 +1015,17 @@ public ExcerptTailer originalToEnd() {
...
 
+    private boolean originalToEndLoopCondition(long approximateLastIndex, long index) {
+        if (direction == FORWARD) {
+            return true;
+        } else if (direction == BACKWARD) {
+            // Do not let index run past the approximate last index
+            return index < approximateLastIndex;
+        } else {
+            return true;
+        }
```
Comment (yevgenp) [inline src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java:1026]: The whole method can be simplified as `if(condition) return smth`. i.e. dropping all `else`.

File: src/test/java/net/openhft/chronicle/queue/impl/single/stress/backwardstailer/BackwardsTailerBoundaryTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    private final RollCycle rollCycle;
+
+    public BackwardsTailerBoundaryTest(RollCycle rollCycle) {
+        this.rollCycle = rollCycle;
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> data() {
+        final List<Object[]> data = new ArrayList<>();
+        data.add(new Object[]{RollCycles.TEST4_DAILY});
```
Comment (yevgenp) [inline src/test/java/net/openhft/chronicle/queue/impl/single/stress/backwardstailer/BackwardsTailerBoundaryTest.java:39]: Not so much parameters :) 
I'd recommend to use a JUnit5 parameterized test (slowly migrating of Junit4)

Comment (tgd) [reply src/test/java/net/openhft/chronicle/queue/impl/single/stress/backwardstailer/BackwardsTailerBoundaryTest.java:39]: :-) Junit5 would have been nicer!

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java (hunk 1 above)
Comment (tgd) [reply src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java:1026]: I agree. I would usually change this, however the code is already integrated into `develop` in this form and we are now backporting to the 2.23 release. I'll be revisiting the code in develop so will address there.

**Pull request merged**